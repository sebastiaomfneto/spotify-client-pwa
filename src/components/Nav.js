import React from "react";

import "./Nav.scss";

function Nav() {
  return (
    <nav>
      <h2>Browse</h2>
      <a href="#/artists">Artists</a>
      <a href="#/album">Album</a>
      <a href="#/track">Track</a>
    </nav>
  );
}

export default Nav;
