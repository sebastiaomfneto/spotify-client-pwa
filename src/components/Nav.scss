.Nav {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-around;
  padding: 1.25rem;
  overflow-y: auto;

  @media (min-width: 768px) {
    flex-direction: column;
  }

  > h2 {
    margin: 0;
    padding-bottom: 0.35rem;
    font-weight: 300;
    font-size: 0.85rem;
    line-height: 1.75rem;
    text-transform: uppercase;
    letter-spacing: 1px;
  }

  > a {
    font-size: 0.85rem;
    color: #b3b3b3;
    text-decoration: none;
    text-overflow: ellipsis;
    line-height: 1.75rem;
    white-space: nowrap;
    &:hover {
      color: #fff;
    }
    &.active {
      &::before {
        content: "";
        position: relative;
        left: -1.125rem;
        border-left: 0.15rem solid #1db954;
      }

      @media (min-width: 768px) {
        width: 8.75rem;
      }
    }
  }
}
