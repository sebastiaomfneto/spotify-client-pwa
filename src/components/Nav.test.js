import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { Nav } from './Nav';

const withProviders = (Element) => <MemoryRouter>{Element}</MemoryRouter>;

describe('testing Nav', () => {
  test('should match with snapshot', () => {
    const { container } = render(withProviders(<Nav />));

    expect(container).toMatchSnapshot();
  });
});
