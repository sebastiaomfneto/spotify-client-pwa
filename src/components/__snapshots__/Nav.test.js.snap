// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Nav should match with snapshot 1`] = `
<div>
  <nav
    class="Nav"
  >
    <h2>
      Browse
    </h2>
    <a
      href="/artists"
    >
      Artists
    </a>
    <a
      href="/albums"
    >
      Albums
    </a>
    <a
      href="/tracks"
    >
      Tracks
    </a>
  </nav>
</div>
`;
